### PR TITLE
:book: Add contributing guide

### PR DESCRIPTION
**Pull request by furkatgofurov7**
Fixes: #4 

The plan is, to have a generic CONTRIBUTING guide for all rancher turtles projects (as of now: `rancher-turtles`, `rancher-turtles-docs`, `rancher-turtles-ui` repos) in this repo and refer to it from other repos (`rancher-turtles`, `rancher-turtles-ui`), which means there are some advanced searching queries I had to use in the guide to refer to things like issues in all 3 repos.

### CODE REVIEW

File: hack/generate-doctoc.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    set -o xtrace
+fi
+
+if [[ -z "$(command -v doctoc)" ]]; then
+  echo "doctoc is not available on your system, skipping verification"
+  echo "Note: The doctoc module can be installed via npm (https://www.npmjs.com/package/doctoc)"
+  exit 0
+fi
+
+doctoc ./CONTRIBUTING.md
```
Comment (richardcase) [inline hack/generate-doctoc.sh:31]:
```suggestion
doctoc --notitle ./CONTRIBUTING.md
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
+# Contributing Guidelines
```
Comment (richardcase) [inline CONTRIBUTING.md:1]: We should think about getting this file included in the generated docs site as well....but we can do that as a follow up.

Comment (furkatgofurov7) [reply CONTRIBUTING.md:1]: I was thinking of moving it over https://github.com/rancher-sandbox/rancher-turtles-docs/tree/main/docs folder, would that make sense?

File: hack/generate-doctoc.sh (hunk 1 above)
Comment (furkatgofurov7) [reply hack/generate-doctoc.sh]: Added, thanks

File: CONTRIBUTING.md (hunk 2 above)
Comment (richardcase) [reply CONTRIBUTING.md:1]: Yeah lets follow up

**Review by richardcase — APPROVED** [review]

**Pull request merged**